pkg/db/database.go

```diff
@@ -54,11 +54,14 @@ func NewPgxDatabase(db *pgxpool.Pool) *PgxDatabase {
 	return &PgxDatabase{db: db}
 }
 
-func (d *PgxDatabase) getLogger() logging.Logger {
+func (d *PgxDatabase) getLogger(ctx context.Context, fields logging.Fields) logging.Logger {
+	var log logging.Logger
 	if d.queryOptions != nil {
-		return d.queryOptions.logger
+		log = d.queryOptions.logger
+	} else {
+		log = logging.Default()
 	}
-	return logging.Default()
+	return log.WithContext(ctx).WithFields(fields)
```

Comment:
General concern - not knowing when to call WithContext - looks like it is something we need to load on our logger as soon as possible or defer it to the last thing we do.

Reply:
I believe this is the right place to do it.  I changed the signature of `getLogger` to say you need to know the context in which you will be logging.  So as soon as you need it, you *must* give context and additional fields.  Now outputting a bad DB log without context and/or fields is actually more difficult than outputting a good one :-)